examples/pytorch/rgcn/entity_classify_mp.py

```diff
@@ -452,14 +452,17 @@ def load_oag(args):
     hg = dataset[0]
 
     # Construct author embeddings by averaging over their papers' embeddings.
-    hg.multi_update_all(
-        {'rev_AP_write_first': (fn.copy_src('emb', 'm'), fn.sum('m', 'h')),
-         'rev_AP_write_last': (fn.copy_src('emb', 'm'), fn.sum('m', 'h')),
-         'rev_AP_write_other': (fn.copy_src('emb', 'm'), fn.sum('m', 'h')),},
-        'sum')
-    cnts = hg.in_degrees(etype='rev_AP_write_first') + hg.in_degrees(etype='rev_AP_write_last') + hg.in_degrees(etype='rev_AP_write_other')
-    cnts = cnts.reshape(-1, 1)
-    hg.nodes['author'].data['emb'] = hg.nodes['author'].data['h'] / cnts
+
+    # make a copy of hg without feature data
+    hg_no_feats = hg.clone()
+    for ntype in hg_no_feats.ntypes:
+        hg_no_feats.nodes[ntype].data.clear()
+    for etype in hg_no_feats.canonical_etypes:
+        hg_no_feats.edges[etype].data.clear()
+    paper_author_subg = hg_no_feats['paper', : , 'author']
+    paper_author_subg.nodes['paper'].data['emb'] = hg.nodes['paper'].data['emb']
+    paper_author_subg.update_all(fn.copy_src('emb', 'm'), fn.mean('m', 'h'))
+    hg.nodes['author'].data['emb'] = paper_author_subg.nodes['author'].data['h']
```

Comment:
why do we have to clone the graph structure here?

Reply:
Because relation slice will duplicate the node feature, which is huge.